django/db/models/expressions.py

```diff
@@ -162,6 +162,9 @@ def __rxor__(self, other):
             "Use .bitand(), .bitor(), and .bitxor() for bitwise logical operations."
         )
 
+    def __invert__(self):
+        return NegatedExpression(self)
```

Comment:
I wounder if we should raise a `TypeError` in cases where `not self.conditional`?

Reply:
Hmm, it would make sense logically, but I'm not sure how we can make it work for an `F`-expression.

For instance in `CanBeActivatedModel.objects.update(is_active=~F("is_active"))`, when negating`F("is_active")`, we didn't yet resolve the field and don't know if it's a boolean field/conditional.
Or is there something I'm missing in the way it could work 🤔

Reply:
Right, the check would need to happen at `NegatedExpression.resolve_expression` time to be effective. Maybe better to not be to restrictive for now.

Reply:
I do realize that I'm late to the party and I was the one who suggested the introduction of `NegatedExpression` but I now wonder if simply returning `Q(self, _negated=True)` could have served the same purpose here.

Reply:
We actually hit a similar issue when using a `Q` object, because the `F` expression is not yet resolved.
Therefore in `django.db.models.sql.query.Query.build_filter` we hit the
```
        if hasattr(filter_expr, "resolve_expression"):
            if not getattr(filter_expr, "conditional", False):
                raise TypeError("Cannot filter against a non-conditional expression.")
```
condition when doing `CanBeActivatedModel.objects.update(is_active=~F("is_active"))` 🤔